js/src/ui/DappVouchFor/store.js

```diff
@@ -20,55 +20,70 @@ import { uniq } from 'lodash';
 import Contracts from '~/contracts';
 import { vouchfor as vouchForAbi } from '~/contracts/abi';
 
+let contractPromise = null;
+
 export default class Store {
   @observable vouchers = [];
 
   constructor (api, app) {
     this._api = api;
 
-    const { contentHash } = app;
+    const { contentHash, id } = app;
+
+    this.findVouchers(contentHash, id);
+  }
+
+  async attachContract () {
+    const address = await Contracts.get().registry.lookupAddress('vouchfor');
```

Comment:
Don't lookup registry per store instance and only create a single contract that is used for queries. (next couple of lines after this). Optimisation as discussed in the previous PR.